### PR TITLE
new template - huawei_vrp_display_arp_all

### PR DESCRIPTION
**Pull request by rafaelsagastume**
New Template Pull Request

COMPONENT

    huawei_vrp_display_arp_all.textfsm, .*, huawei_vrp, disp[[lay]] arp all

SUMMARY

New Huawei Template

### CODE REVIEW

File: ntc_templates/templates/huawei_vrp_display_arp_all.textfsm [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Value Required IP_ADDRESS (\S+)
+Value Required MAC_ADDRESS (\S+)
```
Comment (jvanderaa) [inline ntc_templates/templates/huawei_vrp_display_arp_all.textfsm:2]: Looking to match the previous templates such that if you have a multivendor environment that this is normalized data already ahead of having to edit it. There are several templates that should be updated to match, but the predominate number of the templates are using the suggestion.

```suggestion
Value Required ADDRESS (\S+)
Value Required MAC (\S+)
```

File: ntc_templates/templates/huawei_vrp_display_arp_all.textfsm [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+Value EXPIRE (\d+|\s+)
+Value Required TYPE (D\W\d|I\s-)
+Value Required INTERFACE (\S+)
+Value VPN_INSTANCE (\S+|\s+)
+
+Start
+  ^IP\s+ADDRESS\s+MAC\s+ADDRESS
+  ^\s*VLAN\WCEVLAN\s+PVC
+  ^${IP_ADDRESS}\s+${MAC_ADDRESS}\s+${EXPIRE}\s+${TYPE}\s+${INTERFACE}\s+${VPN_INSTANCE} -> Record
```
Comment (jvanderaa) [inline ntc_templates/templates/huawei_vrp_display_arp_all.textfsm:11]:
```suggestion
  ^${ADDRESS}\s+${MAC}\s+${EXPIRE}\s+${TYPE}\s+${INTERFACE}\s+${VPN_INSTANCE} -> Record
```

File: tests/huawei_vrp/display_arp_all/huawei_vrp_display_arp_all.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  - ip_address: "0.0.0.0"
+    mac_address: "00xx-dxxx-8xxx"
+    expire: " "
+    type: "I -"
+    interface: "Vlanif4025"
+    vpn_instance: "MOBILE-OAM"
+  - ip_address: "0.0.0.0"
+    mac_address: "00xx-dxxx-8xxx"
+    expire: "2"
+    type: "D-0"
```
Comment (jvanderaa) [inline tests/huawei_vrp/display_arp_all/huawei_vrp_display_arp_all.yml:36]:
```suggestion
  - ip_address: "0.0.0.0"
    mac_address: "00xx-dxxx-8xxx"
    expire: " "
    type: "I -"
    interface: "Eth-Trunk3"
    vpn_instance: "RCT"
  - ip_address: "0.0.0.0"
    mac_address: "00xx-dxxx-8xxx"
    expire: "16"
    type: "D-0"
    interface: "Eth-Trunk3"
    vpn_instance: "RCT"
  - address: "0.0.0.0"
    mac: "00xx-dxxx-8xxx"
    expire: " "
    type: "I -"
    interface: "Vlanif4024"
    vpn_instance: "MOBILE-NODE-TRAFFIC"
  - address: "0.0.0.0"
    mac: "00xx-dxxx-8xxx"
    expire: "1"
    type: "D-0"
    interface: "Vlanif4024"
    vpn_instance: "MOBILE-NODE-TRAFFIC"
  - address: "0.0.0.0"
    mac: "00xx-dxxx-8xxx"
    expire: " "
    type: "I -"
    interface: "Vlanif4025"
    vpn_instance: "MOBILE-OAM"
  - address: "0.0.0.0"
    mac: "00xx-dxxx-8xxx"
    expire: "2"
    type: "D-0"
```

**Review by jvanderaa — CHANGES_REQUESTED** [review]
Looking to have the normalization done ahead that may be required for multiple vendor environment.

File: tests/huawei_vrp/display_arp_all/huawei_vrp_display_arp_all.yml (hunk 3 above)
Comment (rafaelsagastume) [reply tests/huawei_vrp/display_arp_all/huawei_vrp_display_arp_all.yml]: thanks for the suggestion

File: ntc_templates/templates/huawei_vrp_display_arp_all.textfsm (hunk 2 above)
Comment (rafaelsagastume) [reply ntc_templates/templates/huawei_vrp_display_arp_all.textfsm]: thanks for the suggestion

File: ntc_templates/templates/huawei_vrp_display_arp_all.textfsm (hunk 1 above)
Comment (rafaelsagastume) [reply ntc_templates/templates/huawei_vrp_display_arp_all.textfsm]: thanks for the suggestion

**Pull request closed without merging**